varinfo/var_info.py

```diff
@@ -362,6 +362,39 @@ def get_required_dimensions(self, variables: Set[str]) -> Set[str]:
             if self.get_variable(dimension) is not None
         )
 
+    def get_missing_variable_attributes(self, variable_name: str) -> dict[str, Any]:
+        """Return a single set of all attributes for a variable that is not present
+        in the granule file(e.g. grid_mapping variable in collections that are not
+        fully CF compliant).
+
+        """
+        variable_attributes = self.cf_config.get_cf_attributes(variable_name)
+        return (
+            variable_attributes['cf_supplements'] | variable_attributes['cf_overrides']
+        )
+
+    def get_references_for_attribute(
+        self, list_of_variables: list[str], reference_attribute_name: str
+    ) -> set[str]:
+        """Return a single set of all references in a specific metadat attribute
+        for a list of variables (e.g. bounds, coordinates, cf attributes).
+
+        """
+        # Iterate through all requested variables and extract a list of
+        # references for the metadata attribute. This will produce a list of lists,
+        # which should be flattened into a single list and then combined into a set
+        # to remove duplicates.
+        reference_list = [
+            list(self.get_variable(variable).references.get(reference_attribute_name))
+            for variable in list_of_variables
+            if self.get_variable(variable).references.get(reference_attribute_name)
+            is not None
+        ]
+        all_references = [
+            reference for references in reference_list for reference in references
+        ]
+        return set(all_references)
```

Comment:
I think you can simplify these two lines by doing a set comprehension, rather than a list comprehension and casting it to a set:

```
return set(reference for references in reference_list for reference in references)
```

Also, given that ordering in a double set/list comprehension is sometimes hard to grok, maybe a couple of the variable names in this comprehension could be renamed. I'm not sure if this is clearer, but how about something like:

```
return set(
    variable_reference
    for variable_references in reference_list
    for variable_reference in variable_references
)
```

Reply:
is this close to what you think is needed?

r```
eference_set = [
            self.get_variable(variable).references.get(reference_attribute_name)
            for variable in list_of_variables
            if self.get_variable(variable).references.get(reference_attribute_name)
            is not None
        ]
        return set(
            variable_reference
            for variable_references in reference_set
            for variable_reference in variable_references
        )
```

Reply:
Yup, that's what I was thinking. Thanks.